light_curves/code_src/hats_functions.py

```diff
@@ -0,0 +1,133 @@
+    err_col : str
+        Name of the flux-uncertainty column in the light-curve catalog.
+    radius : float, optional
+        Cross-match radius in arcseconds (default 1.0).
+
+    Returns
+    -------
+    MultiIndexDFObject
+        A multi-indexed (objectid, label, band, time) container of the light curves.
+    """
+    # 1) read the two catalogs lazily
+    
+    # this table will be used for cross matching with our sample's ra and decs
+    # but does not have light curve information
+    obj_cat = lsdb.read_hats(object_catalog_path,
+                             columns=object_columns,
+                             margin_cache=object_margin_cache)
+    # this table houses the light curves
+    lc_cat  = lsdb.read_hats(lightcurve_catalog_path,
+                             columns=light_curve_columns,
+                            margin_cache=lightcurve_margin_cache)
+    
+    # 2) convert astropy sample_table into an LSDB catalog
+    sample_df = pd.DataFrame({
+        'objectid': sample_table['objectid'],
+        'ra_deg':    sample_table['coord'].ra.deg,
+        'dec_deg':   sample_table['coord'].dec.deg,
+        'label':     sample_table['label']
+    })
+    # convert dataframe to hats catalog
+    sample_lsdb = lsdb.from_dataframe(
+        sample_df,
+        ra_column='ra_deg',
+        dec_column='dec_deg',
+        margin_threshold=10,
+        drop_empty_siblings=True
+    )
+
+    # 3) cross-match to find nearest object
+    # only keep the best match
+    matched_obj = obj_cat.crossmatch(
+        sample_lsdb,
+        radius_arcsec=radius,
+        n_neighbors=1,
+        suffixes=("", "")     # ← prevent “objID” → “objID_1”
+    )
+
+    # 4) join to the light-curve catalog
+    matched_lc = matched_obj.join(
+        lc_cat,
+        left_on=id_col,
+        right_on=id_col,
+        output_catalog_name=f"{id_col}_lc",
+        suffixes=["", ""]
+    )
+
+    # 5) compute on Dask
+    # here is where the actual work gets done
+    with Client():
+        matched_df = matched_lc.compute()
+
+    # 6) handle no matches
+    if matched_df.empty:
+        return MultiIndexDFObject(data=pd.DataFrame())
+
+    # 7) map filter IDs to names
+    filter_col = next(c for c in light_curve_columns if 'filter' in c.lower())
+    band_names = np.vectorize(filter_id_to_name.get)(matched_df[filter_col])
+
+    # 8) build and index the DataFrame
+    df_lc = pd.DataFrame({
+        'flux':     pd.to_numeric(matched_df[flux_col], errors='coerce'),
+        'err':      pd.to_numeric(matched_df[err_col],  errors='coerce'),
+        'time':     pd.to_numeric(matched_df[time_col], errors='coerce'),
+        'objectid': matched_df['objectid'].astype(np.int64),
+        'band':     band_names,
+        'label':    matched_df['label'].astype(str)
+    })
+    
+    df = pd.DataFrame(df_lc).set_index(['objectid', 'label', 'band', 'time'])
+    
+    return MultiIndexDFObject(data=df)
```

Comment:
```suggestion
    df_lc = df_lc.set_index(['objectid', 'label', 'band', 'time'])
    
    return MultiIndexDFObject(data=df_lc)
```

Use the same name to avoid making an extra copy.